clusterDeployer.py

```diff
@@ -199,6 +200,20 @@ def teardown(self) -> None:
 
         print(lh.run(f"ip link set eno1 nomaster"))
 
+    def _validate_external_port(self, lh) -> Optional[str]:
+        # do automatic detection, if needed
+        if self._cc["external_port"] == "auto":
+            self._cc["external_port"] = lh.port_from_route("default")
+            if not self._cc["external_port"]:
+                 return None
```

Comment:
function

Reply:
I don't follow ?!? this will not be re-usable and moving the above 4 lines into a separate function (where? under cluserDeployer?) will make the code more fragmented and IMHO less readable

Reply:
Function are not (only) about reuse, but also about readability. The fact that you put a comment above these lines explaining what this does, raises a red flag for me. The issue with that comment is that it might get out of sync with the code. At the same time, the comment does add value. For that reason, I suggest creating a function with a name that is so descriptive that we don't need a comment.

Reply:
I guess we have different opinions here. I need to use a function name almost as long as the commit to make it as descriptive and at the that point I prefer a comment: easier to read, and overall more compact. Anyway your call.